unit/pointer-analysis/value_set.cpp

```diff
@@ -47,29 +47,24 @@ SCENARIO(
   {
     // Create struct A { A *x; A *y }
 
-    struct_typet struct_A;
+    struct_typet struct_A({{"x", pointer_typet(symbol_typet("A"), 64)},
+                           {"y", pointer_typet(symbol_typet("A"), 64)}});
     struct_A.set_tag("A");
```

Comment:
These should now actually be struct_tag types!

Reply:
If that's a strict requirement in goto programs then we need a validation pass that says so?! Otherwise that unit test would seem ok to me?